### ISSUE TITLE
Preserve On Refresh does not load JS files after refresh or redirects from external url.

### ISSUE THREAD

**Issue by abdullahtellioglu** [vaadin/flow#16061]
### Description of the bug

I have a page which is annotated via `@PreserveOnRefresh` also has js files included . In the first run when I execute js functions in the file everything works fine. Executing JS functions do not work after refreshing the page or redirecting to external url and returning back.

![preserve-on-load-ex2](https://user-images.githubusercontent.com/100126447/221592666-20289510-9437-43b2-8c8d-f1407747f303.png)
![preserve-on-load-ex1](https://user-images.githubusercontent.com/100126447/221592689-c1e34d59-d377-46c7-8457-ed0e0890f68f.png)

### Expected behavior

Added JS files should be stored in DOM or session after reloading the page. 

### Minimal reproducible example

```

@Route("preserve-on-refresh")
@PreserveOnRefresh
public class PreserveOnRefreshTest extends VerticalLayout {
    public PreserveOnRefreshTest(){
        Button executeLogoutBtn = new Button("Execute logout");
        executeLogoutBtn.addClickListener(event -> {
           UI.getCurrent().getPage().executeJs("logout();");
        });
        Anchor openExternalUrl = new Anchor("https://www.google.com", "External link");
        add(executeLogoutBtn);
        add(openExternalUrl);
        UI.getCurrent().getPage().addJavaScript("context://js/logout.js");
    }
}
```
JS file in `webapp/js/logout.js`

```
function logout() {
    console.log("Logout has clicked");
    alert("Logout has clicked");
}
```

### Versions

- Vaadin / Flow version: 23.3.2
- Java version: 11
- OS version: 
- Browser version (if applicable):
- Application Server (if applicable):
- IDE (if applicable):

**Comment by mcollovati** [comment]
With `@PreserveOnRefresh` the component is not recreated on reload, but the UI is.
So, when you reload the page, the javascript is not added to the new UI, since the constructor is not invoked.
You can try to move the code in the `onAttach` method or in an attach listener.

**Comment by abdullahtellioglu** [comment]
Moving the code in the `onAttach` works. Although the expectation is keeping everything same if page is annotated as `@PreserveOnRefresh`

**Comment by mcollovati** [comment]
There is a limitations section on documentation reporting this behavior: https://vaadin.com/docs/latest/advanced/preserving-state-on-refresh#preconditions-and-limitations

Do you think the docs should be improved to make it more evident?

**Comment by abdullahtellioglu** [comment]
Yes, I think documentation needs to be improved with situation like these otherwise it seems like a bug.
